### PR TITLE
feat: [HL-2] Add Factory package + Example 💉

### PR DESCRIPTION
**Pull request by mhmdatallaa**
### Description
- Add Factory package
- Add Example in README.md file

### CODE REVIEW

**Comment by ahmdmhasn** [comment]
I will move it to draft until it's ready 🚀
@MohamedAtallaa

File: README.md [hunk 1]
```diff
@@ -4,6 +4,113 @@ Motoon Mentorship Program Team Educational Project
...
+    }
+    
+    var homeViewModel: Factory<HomeViewModel> {
+        Factory(self) { HomeViewModel()
+        }
+    }
+}
+```
+
 ## Links 🔗
```
Comment (ahmdmhasn) [inline README.md]: Please move the links to the top

File: README.md [hunk 2]
```diff
@@ -4,6 +4,113 @@ Motoon Mentorship Program Team Educational Project
...
+    var meal: Factory<Meal> {
+        Factory(self) { Meal(name: "Omlet", ingrediants: ["Salt", "Oil", "2 Eggs"]) }
+    }
+    
+    var homeViewModel: Factory<HomeViewModel> {
+        Factory(self) { HomeViewModel()
+        }
+    }
+}
+```
```
Comment (ahmdmhasn) [inline README.md]:
```suggestion
```

This part has a lot of duplications and few invalid points like we don't normally register the model for reusability. I suggest to keep the first section which is how to use it until we agree on the architecture. 

Thanks for your efforts

**Review by ahmdmhasn — APPROVED** [review]

**Review by ahmednasr8898 — APPROVED** [review]

**Review by ahmdyasser — APPROVED** [review]

**Review by ahmdmhasn — APPROVED** [review]
🚀🚀

**Pull request merged**